test/functional/output_kafka.go

```diff
@@ -0,0 +1,85 @@
+package functional
+
+import (
+	"github.com/ViaQ/logerr/log"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/helpers/kafka"
+	"github.com/openshift/cluster-logging-operator/test/runtime"
+	"strings"
+)
+
+const (
+	ImageRemoteKafka   = kafka.KafkaImage
+	OpenshiftLoggingNS = "openshift-logging"
+)
+
+func (f *FluentdFunctionalFramework) addKafkaOutput(b *runtime.PodBuilder, output logging.OutputSpec) error {
+	log.V(2).Info("Adding kafka output", "name", output.Name)
+	name := strings.ToLower(output.Name)
+
+	log.V(2).Info("Standing up Kafka instance", "name", name)
+	//steps to deploy kafka single node cluster a. Deploy Zookeeper b. Deploy Broker c. Deploy kafka Consumer
+
+	//step a
+	// Deploy Zookeeper steps : create configmap, create statefulset?, create zookeeper service
+	zookeepercm := kafka.NewZookeeperConfigMap(OpenshiftLoggingNS)
+	log.V(2).Info("Creating zookeeper configmap", "namespace", zookeepercm.Namespace, "name", zookeepercm.Name)
+	if err := f.Test.Client.Create(zookeepercm); err != nil {
+		return err
+	}
+	zookeeperapp := kafka.NewZookeeperStatefuleSet(OpenshiftLoggingNS)
+	log.V(2).Info("Creating zookeeper StatefulSet", "namespace", zookeeperapp.Namespace, "name", zookeeperapp.Name)
+	if err := f.Test.Client.Create(zookeeperapp); err != nil {
+		return err
+	}
+	zookeepersvc := kafka.NewZookeeperService(OpenshiftLoggingNS)
```

Comment:
Similar to the statefulset comment, you may not require a service since the container will live in the same pod as the functional test.  It may require config changes that would be reflected in the configmap